.github/workflows/test.yml

```
@@ -0,0 +1,31 @@
+name: Test
+
+on:
+  push:
+    branches: ["master"]
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+        fail-fast: false
+        max-parallel: 1
+        matrix:
+            python-version: ["3.7", "3.8", "3.9", "3.10"]
```

```suggestion
            python-version: ["3.7", "3.8", "3.9"]
```

3.10 is not expected to work yet (because of nose) so lets not add it until we know it passes. It may be a bit academic since the other tests are also failing, but at least they should be passing.

---

I confirmed I can test even I use unittest instead of nose. So I replace nose with unittest.